File: erpcloud_itagksa/itag_ksa_buying/supplier/supplier.py
# -*- coding: utf-8 -*-
# Copyright (c) 2026, ITAG and contributors
# For license information, please see license.txt

from frappe.utils import add_to_date, getdate


def validate(doc, method=None):
	set_next_reevaluation_date(doc)


def set_next_reevaluation_date(doc):
	"""Compute Next Re-Evaluation Date = base + frequency years.

	Base is the last re-evaluation date, falling back to the initial approval
	date when the supplier has never been re-evaluated. Frequency is the Select
	label ("1 Year", "2 years", ...) — the leading integer is the year count.
	"""
	base = doc.custom_last_reevaluation_date or doc.custom_initial_approval_date
	if not base or not doc.custom_frequency_of_reevaluation:
		return

	years = int(doc.custom_frequency_of_reevaluation.split()[0])
	doc.custom_next_reevaluation_date = add_to_date(getdate(base), years=years)

File: erpcloud_itagksa/itag_ksa_buying/supplier/supplier.js
// Copyright (c) 2026, ITAG and contributors
// For license information, please see license.txt

frappe.ui.form.on('Supplier', {
	refresh: function (frm) {
		frappe.db.get_single_value('ITAG KSA Settings', 'reevaluation_role').then((role) => {
			const allowed = !role || (frappe.user_roles || []).includes(role);
			frm.set_df_property('custom_supplier_approval', 'hidden', allowed ? 0 : 1);
		});
	},
});
